docs/source/run/parameters.rst

```diff
@@ -747,6 +747,11 @@ Field diagnostics
     Otherwise only ``rhomjz`` equal to :math:`\rho-j_z/c` will be available.
     If ``rho`` is explicitly mentioned in ``diagnostic.field_data``, then the default will become `1`.
 
+* ``hipace.deposit_rho_individual`` (`bool`) optional (default `0`)
+    This option works similar to ``hipace.deposit_rho``,
+    however the charge density from every plasma species will be deposited into individual fields
+    that are accessible as ``rho_<plasma name>`` in ``diagnostic.field_data``.
+
```

Comment:
Users would look for per-species rho for the sake of diagnostics (they don't care whether this requires a specific deposition or not), so I believe the main option for them to specify this would be as `<diag_name>.field_data`. Could you add an option there? I believe letting the user select the species `rho_electrons`, `rho_ions` in `field_data` could be nice but probably overkill, so a keyword like `rho_per_species` would work great. Could you add this in the doc and the handling in the code? The option above `hipace.deposit_rho_individual` can stay, but I could imagine that it's more for developers.